File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/Ordenes.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports CapaNegocio
Imports System.Data.SqlClient

Public Partial Class Ordenes
	Inherits Form
	Private numeroPaginaPedidosPendientes As Integer = 1
	Private numeroPaginaPedidosEntegados As Integer = 1
	Private registrosPorPagina As Integer = 25
	Private cantidadPaginasPedidosPendientes As Integer
	Private cantidadPaginasPedidosEntregados As Integer

	Private esPendiente As Integer = 1

	Public Sub New()
		InitializeComponent()
		Mostrar()
	End Sub

	Private Sub tabControlPedidos_Selected(sender As Object, e As TabControlEventArgs)
		If tabControlPedidos.SelectedTab Is tabControlPedidos.TabPages("tabPagePendientes") Then
			esPendiente = 1
			Mostrar()
		ElseIf tabControlPedidos.SelectedTab Is tabControlPedidos.TabPages("tabPageEntregados") Then
			esPendiente = 0
			Mostrar()
		End If
	End Sub

	Public Sub Mostrar()
		Try
			If esPendiente = 1 Then
				Me.Dock = DockStyle.Fill
				Me.dataGridViewPedidosPendientes.DataSource = NOrdenes.MostrarOrdenesPendientes(numeroPaginaPedidosPendientes, registrosPorPagina)
				Me.dataGridViewPedidosPendientes.Columns(1).Visible = False
				Me.dataGridViewPedidosPendientes.Columns(8).DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight
				Me.dataGridViewPedidosPendientes.Columns(8).DefaultCellStyle.Format = "0.00## $"
				cantidadPaginasPedidosPendientes = NOrdenes.TamañoOrdenesPendientes(registrosPorPagina)
				Me.labelPagina.Text = [String].Format("Página {0} de {1}", numeroPaginaPedidosPendientes, cantidadPaginasPedidosPendientes)

			ElseIf esPendiente = 0 Then
				Me.Dock = DockStyle.Fill
				Me.dataGridViewPedidosEntregados.DataSource = NOrdenes.MostrarOrdenesEntregados(numeroPaginaPedidosEntegados, registrosPorPagina)
				Me.dataGridViewPedidosEntregados.Columns(1).Visible = False
				Me.dataGridViewPedidosEntregados.Columns(8).DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight
				Me.dataGridViewPedidosEntregados.Columns(8).DefaultCellStyle.Format = "0.00## $"
				cantidadPaginasPedidosEntregados = NOrdenes.TamañoOrdenesEntregados(registrosPorPagina)
				Me.labelPagina.Text = [String].Format("Página {0} de {1}", numeroPaginaPedidosEntegados, cantidadPaginasPedidosEntregados)
			End If

		Catch ex As Exception
			MensajeError(ex.Message)
		End Try
	End Sub

	Public Sub Buscar()
		Try
			If esPendiente = 1 Then
				Me.dataGridViewPedidosPendientes.DataSource = NOrdenes.BuscarOrdenPendiente(Me.textBoxBuscarNombre.Text)
			Else
				Me.dataGridViewPedidosEntregados.DataSource = NOrdenes.BuscarOrdenEntregado(Me.textBoxBuscarNombre.Text)
			End If
		Catch ex As Exception
			MensajeError(ex.Message)
		End Try
	End Sub

	Public Sub Refrescar()
		If esPendiente = 1 Then
			Me.numeroPaginaPedidosPendientes = 1
		ElseIf esPendiente = 0 Then
			Me.numeroPaginaPedidosEntegados = 1
		End If

		Me.Mostrar()
		Me.textBoxBuscarNombre.Text = [String].Empty
	End Sub

	Public Sub Mensaje(mensaje__1 As [String])
		Me.labelMensajes.Text = mensaje__1
	End Sub

	Public Sub LimpiarMensaje()
		Me.labelMensajes.Text = [String].Empty
	End Sub

	Public Sub MensajeError(mensaje As String)
		MessageBox.Show(mensaje, "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
	End Sub

	Public Function ObtenerSeleccionPedidos() As DataGridViewRow
		Dim filaSeleccionada As DataGridViewRow = Nothing

		If esPendiente = 1 Then
			filaSeleccionada = Me.dataGridViewPedidosPendientes.Rows(Me.dataGridViewPedidosPendientes.CurrentRow.Index)

		ElseIf esPendiente = 0 Then
			filaSeleccionada = Me.dataGridViewPedidosEntregados.Rows(Me.dataGridViewPedidosEntregados.CurrentRow.Index)
		End If

		Return filaSeleccionada
	End Function

	Private Sub textBoxBuscarNombre_TextChanged(sender As Object, e As EventArgs)
		If Me.textBoxBuscarNombre.Text = [String].Empty Then
			If esPendiente = 1 Then
				Me.numeroPaginaPedidosPendientes = 1
			ElseIf esPendiente = 0 Then
				Me.numeroPaginaPedidosEntegados = 1
			End If

			Me.Mostrar()
			Me.tableLayoutPanelPaginacion.Show()
		Else
			Me.Buscar()
			Me.tableLayoutPanelPaginacion.Hide()
		End If
	End Sub

	Private Sub buttonRefrescar_Click(sender As Object, e As EventArgs)
		Me.Refrescar()
		Me.LimpiarMensaje()
	End Sub

	Private Sub buttonEditar_Click(sender As Object, e As EventArgs)
		Try
			If esPendiente = 1 Then
				If Me.dataGridViewPedidosPendientes.Rows.Count > 0 Then
					Dim editarPedidoPendiente As New FrmEditarOrdenPendiente(Me)
					editarPedidoPendiente.ShowDialog()
				Else
					MensajeError("Debes seleccionar una fila para editar")
				End If

			ElseIf esPendiente = 0 Then
				If Me.dataGridViewPedidosEntregados.Rows.Count > 0 Then
					Dim editarPedidoEntregado As New FrmEditarOrdenEntregado(Me)
					editarPedidoEntregado.ShowDialog()
				Else
					MensajeError("Debes seleccionar una fila para editar")
				End If
			End If

		Catch ex As Exception
			MensajeError(ex.Message)
		End Try
	End Sub

	Private Sub Eliminar()
		Dim confirmacion As DialogResult = MessageBox.Show("¿Seguro deseas eliminar esta Orden?", "Eliminar Orden", MessageBoxButtons.OKCancel, MessageBoxIcon.Information)

		If confirmacion = DialogResult.OK Then
			Dim mensaje__1 As [String] = NOrdenes.Eliminar(Convert.ToInt32(ObtenerSeleccionPedidos().Cells("ID").Value))
			If mensaje__1 = "Y" Then
				Mensaje([String].Format("El Pedido {0} de {1} ha sido ELIMINADO", Convert.ToString(ObtenerSeleccionPedidos().Cells("CÓDIGO").Value), Convert.ToString(ObtenerSeleccionPedidos().Cells("CLIENTE").Value)))
				Refrescar()
			Else
				MensajeError(mensaje__1)
				Refrescar()
			End If
		End If
	End Sub

	Private Sub buttonEliminar_Click(sender As Object, e As EventArgs)
		Try
			If esPendiente = 1 Then
				If Me.dataGridViewPedidosPendientes.Rows.Count > 0 Then
					Me.Eliminar()
				Else
					MensajeError("Debes seleccionar una fila para eliminar")
				End If
			ElseIf esPendiente = 0 Then
				If Me.dataGridViewPedidosEntregados.Rows.Count > 0 Then
					Me.Eliminar()
				Else
					MensajeError("Debes seleccionar una fila para eliminar")
				End If
			End If

		Catch ex As Exception
			MensajeError(ex.Message)
		End Try
	End Sub

	Private Sub buttonAgregar_Click(sender As Object, e As EventArgs)
		Try
			Dim nuevoPedido As New FrmAgregarNuevaOrden(Me)
			nuevoPedido.ShowDialog()
		Catch ex As Exception
			MensajeError(ex.Message)
		End Try
	End Sub

	Private Sub dataGridViewPedidosPendientes_CellContentClick(sender As Object, e As DataGridViewCellEventArgs)
		Dim senderGrid = DirectCast(sender, DataGridView)

		If TypeOf senderGrid.Columns(e.ColumnIndex) Is DataGridViewButtonColumn AndAlso e.RowIndex >= 0 Then
			Try
				Dim pedidoDetalles As New FrmoOrdenPendienteDetalles(Me)
				pedidoDetalles.ShowDialog()
			Catch ex As Exception
				MensajeError(ex.Message)
			End Try
		End If
	End Sub

	Private Sub dataGridViewPedidosEntregados_CellContentClick(sender As Object, e As DataGridViewCellEventArgs)
		Dim senderGrid = DirectCast(sender, DataGridView)

		If TypeOf senderGrid.Columns(e.ColumnIndex) Is DataGridViewButtonColumn AndAlso e.RowIndex >= 0 Then
			Try
				Dim pedidoEntregadoDetalles As New FrmOrdenEntregadoDetalles(Me)
				pedidoEntregadoDetalles.ShowDialog()
			Catch ex As Exception
				MensajeError(ex.Message)
			End Try
		End If
	End Sub

	Private Sub buttonMarcarEntregado_Click(sender As Object, e As EventArgs)
		Try
			If esPendiente = 1 Then
				If Me.dataGridViewPedidosPendientes.Rows.Count > 0 Then
					Dim marcarEntregado As New FrmMarcarEntregado(Me)
					marcarEntregado.ShowDialog()
				Else
					MensajeError("Debes seleccionar una fila para marcar como Entregado")
				End If

			ElseIf esPendiente = 0 Then
				If Me.dataGridViewPedidosEntregados.Rows.Count > 0 Then
					Dim marcarPendiente As New FrmMarcarPendiente(Me)
					marcarPendiente.ShowDialog()
				Else
					MensajeError("Debes seleccionar una fila para marcar como Pendiente")
				End If
			End If

		Catch ex As Exception
			MensajeError(ex.Message)
		End Try

	End Sub

	Private Sub buttonAnterior_Click(sender As Object, e As EventArgs)
		If numeroPaginaPedidosPendientes > 1 AndAlso esPendiente = 1 Then
			numeroPaginaPedidosPendientes = numeroPaginaPedidosPendientes - 1
			Mostrar()
		ElseIf numeroPaginaPedidosEntegados > 1 AndAlso esPendiente = 0 Then
			numeroPaginaPedidosEntegados = numeroPaginaPedidosEntegados - 1
			Mostrar()
		End If
	End Sub

	Private Sub buttonSiguiente_Click(sender As Object, e As EventArgs)
		If numeroPaginaPedidosPendientes < cantidadPaginasPedidosPendientes AndAlso esPendiente = 1 Then
			numeroPaginaPedidosPendientes = numeroPaginaPedidosPendientes + 1
			Mostrar()
		ElseIf numeroPaginaPedidosEntegados < cantidadPaginasPedidosEntregados AndAlso esPendiente = 0 Then
			numeroPaginaPedidosEntegados = numeroPaginaPedidosEntegados + 1
			Mostrar()
		End If
	End Sub

End Class

File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/Ordenes.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using CapaNegocio;
using System.Data.SqlClient;

namespace CapaPresentacion
{
    public partial class Ordenes : Form
    {
        int numeroPaginaPedidosPendientes = 1;
        int numeroPaginaPedidosEntegados = 1;
        int registrosPorPagina = 25;
        int cantidadPaginasPedidosPendientes;
        int cantidadPaginasPedidosEntregados;

        int esPendiente = 1;
        public Ordenes()
        {
            InitializeComponent();
            Mostrar();
        }

        private void tabControlPedidos_Selected(object sender, TabControlEventArgs e)
        {
            if (tabControlPedidos.SelectedTab == tabControlPedidos.TabPages["tabPagePendientes"])
            {
                esPendiente = 1;
                Mostrar();
            }
            else if (tabControlPedidos.SelectedTab == tabControlPedidos.TabPages["tabPageEntregados"])
            {
                esPendiente = 0;
                Mostrar();
            }
        }

        public void Mostrar()
        {
            try
            {
                if (esPendiente == 1)
                {
                    this.Dock = DockStyle.Fill;
                    this.dataGridViewPedidosPendientes.DataSource = NOrdenes.MostrarOrdenesPendientes(numeroPaginaPedidosPendientes, registrosPorPagina);
                    this.dataGridViewPedidosPendientes.Columns[1].Visible = false;
                    this.dataGridViewPedidosPendientes.Columns[8].DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight;
                    this.dataGridViewPedidosPendientes.Columns[8].DefaultCellStyle.Format = "0.00## Q";
                    cantidadPaginasPedidosPendientes = NOrdenes.TamañoOrdenesPendientes(registrosPorPagina);
                    this.labelPagina.Text = String.Format("Página {0} de {1}", numeroPaginaPedidosPendientes, cantidadPaginasPedidosPendientes);
                }

                else if (esPendiente == 0)
                {
                    this.Dock = DockStyle.Fill;
                    this.dataGridViewPedidosEntregados.DataSource = NOrdenes.MostrarOrdenesEntregados(numeroPaginaPedidosEntegados, registrosPorPagina);
                    this.dataGridViewPedidosEntregados.Columns[1].Visible = false;
                    this.dataGridViewPedidosEntregados.Columns[8].DefaultCellStyle.Alignment = DataGridViewContentAlignment.MiddleRight;
                    this.dataGridViewPedidosEntregados.Columns[8].DefaultCellStyle.Format = "0.00## Q";
                    cantidadPaginasPedidosEntregados = NOrdenes.TamañoOrdenesEntregados(registrosPorPagina);
                    this.labelPagina.Text = String.Format("Página {0} de {1}", numeroPaginaPedidosEntegados, cantidadPaginasPedidosEntregados);
                }
            }

            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        public void Buscar()
        {
            try
            {
                if (esPendiente == 1)
                {
                    this.dataGridViewPedidosPendientes.DataSource = NOrdenes.BuscarOrdenPendiente(this.textBoxBuscarNombre.Text);
                }
                else
                {
                    this.dataGridViewPedidosEntregados.DataSource = NOrdenes.BuscarOrdenEntregado(this.textBoxBuscarNombre.Text);
                }
            }
            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        public void Refrescar()
        {
            if (esPendiente == 1)
            {
                this.numeroPaginaPedidosPendientes = 1;
            }
            else if (esPendiente == 0)
            {
                this.numeroPaginaPedidosEntegados = 1;
            }

            this.Mostrar();
            this.textBoxBuscarNombre.Text = String.Empty;
        }

        public void Mensaje(String mensaje)
        {
            this.labelMensajes.Text = mensaje;
        }

        public void LimpiarMensaje()
        {
            this.labelMensajes.Text = String.Empty;
        }

        public void MensajeError(string mensaje)
        {
            MessageBox.Show(mensaje, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
        }

        public DataGridViewRow ObtenerSeleccionPedidos()
        {
            DataGridViewRow filaSeleccionada = null;

            if (esPendiente == 1)
            {
                filaSeleccionada = this.dataGridViewPedidosPendientes.Rows[this.dataGridViewPedidosPendientes.CurrentRow.Index];
            }

            else if (esPendiente == 0)
            {
                filaSeleccionada = this.dataGridViewPedidosEntregados.Rows[this.dataGridViewPedidosEntregados.CurrentRow.Index];
            }

            return filaSeleccionada;
        }

      

        private void textBoxBuscarNombre_TextChanged(object sender, EventArgs e)
        {
            if (this.textBoxBuscarNombre.Text == String.Empty)
            {
                if (esPendiente == 1)
                {
                    this.numeroPaginaPedidosPendientes = 1;
                }
                else if (esPendiente == 0)
                {
                    this.numeroPaginaPedidosEntegados = 1;
                }

                this.Mostrar();
            }
            else
            {
                this.Buscar();
            }
        }

        private void pictureBox5_Click(object sender, EventArgs e)
        {
            this.Refrescar();
            this.LimpiarMensaje();
        }

        private void pictureBox3_Click(object sender, EventArgs e)
        {
            try
            {
                if (esPendiente == 1)
                {
                    if (this.dataGridViewPedidosPendientes.Rows.Count > 0)
                    {
                        FrmEditarOrdenPendiente editarPedidoPendiente = new FrmEditarOrdenPendiente(this);
                        editarPedidoPendiente.ShowDialog();
                    }
                    else
                    {
                        MensajeError("Debes seleccionar una fila para editar");
                    }
                }

                else if (esPendiente == 0)
                {
                    if (this.dataGridViewPedidosEntregados.Rows.Count > 0)
                    {
                        FrmEditarOrdenEntregado editarPedidoEntregado = new FrmEditarOrdenEntregado(this);
                        editarPedidoEntregado.ShowDialog();
                    }
                    else
                    {
                        MensajeError("Debes seleccionar una fila para editar");
                    }
                }
            }

            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        private void Eliminar()
        {
            DialogResult confirmacion = MessageBox.Show("¿Seguro deseas eliminar esta Orden?", "Eliminar Orden",
            MessageBoxButtons.OKCancel, MessageBoxIcon.Information);

            if (confirmacion == DialogResult.OK)
            {
                String mensaje = NOrdenes.Eliminar(Convert.ToInt32(ObtenerSeleccionPedidos().Cells["ID"].Value));
                if (mensaje == "Y")
                {
                    Mensaje(String.Format("El Pedido {0} de {1} ha sido ELIMINADO",
                        Convert.ToString(ObtenerSeleccionPedidos().Cells["CÓDIGO"].Value),
                        Convert.ToString(ObtenerSeleccionPedidos().Cells["PROVEEDOR"].Value)));
                    Refrescar();
                }
                else
                {
                    MensajeError(mensaje);
                    Refrescar();
                }
            }
        }

        private void pictureBox2_Click(object sender, EventArgs e)
        {
            try
            {
                if (esPendiente == 1)
                {
                    if (this.dataGridViewPedidosPendientes.Rows.Count > 0)
                    {
                        this.Eliminar();
                    }
                    else
                    {
                        MensajeError("Debes seleccionar una fila para eliminar");
                    }
                }
                else if (esPendiente == 0)
                {
                    if (this.dataGridViewPedidosEntregados.Rows.Count > 0)
                    {
                        this.Eliminar();
                    }
                    else
                    {
                        MensajeError("Debes seleccionar una fila para eliminar");
                    }
                }
            }

            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        private void pictureBox4_Click(object sender, EventArgs e)
        {
            try
            {
                FrmAgregarNuevaOrden nuevoPedido = new FrmAgregarNuevaOrden(this);
                nuevoPedido.ShowDialog();
            }
            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        private void dataGridViewPedidosPendientes_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {
            var senderGrid = (DataGridView)sender;

            if (senderGrid.Columns[e.ColumnIndex] is DataGridViewButtonColumn &&
                e.RowIndex >= 0)
            {
                try
                {
                    FrmOrdenPendienteDetalles pedidoDetalles = new FrmOrdenPendienteDetalles(this);
                    pedidoDetalles.ShowDialog();
                }
                catch (Exception ex)
                {
                    MensajeError(ex.Message);
                }
            }
        }

        private void dataGridViewPedidosEntregados_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {
            var senderGrid = (DataGridView)sender;

            if (senderGrid.Columns[e.ColumnIndex] is DataGridViewButtonColumn &&
                e.RowIndex >= 0)
            {
                try
                {
                    FrmOrdenEntregadoDetalles pedidoEntregadoDetalles = new FrmOrdenEntregadoDetalles(this);
                    pedidoEntregadoDetalles.ShowDialog();
                }
                catch (Exception ex)
                {
                    MensajeError(ex.Message);
                }
            }
        }

        private void pictureBox7_Click(object sender, EventArgs e)
        {
            try
            {
                if (esPendiente == 1)
                {
                    if (this.dataGridViewPedidosPendientes.Rows.Count > 0)
                    {
                        FrmMarcarEntregado marcarEntregado = new FrmMarcarEntregado(this);
                        marcarEntregado.ShowDialog();
                    }
                    else
                    {
                        MensajeError("Debes seleccionar una fila para marcar como Entregado");
                    }
                }

                else if (esPendiente == 0)
                {
                    if (this.dataGridViewPedidosEntregados.Rows.Count > 0)
                    {
                        FrmMarcarPendiente marcarPendiente = new FrmMarcarPendiente(this);
                        marcarPendiente.ShowDialog();
                    }
                    else
                    {
                        MensajeError("Debes seleccionar una fila para marcar como Pendiente");
                    }
                }
            }

            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        private void pictureBox6_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void dataGridViewPedidosEntregados_CellContentClick_1(object sender, DataGridViewCellEventArgs e)
        {
            var senderGrid = (DataGridView)sender;

            if (senderGrid.Columns[e.ColumnIndex] is DataGridViewButtonColumn &&
                e.RowIndex >= 0)
            {
                try
                {
                    FrmOrdenEntregadoDetalles pedidoEntregadoDetalles = new FrmOrdenEntregadoDetalles(this);
                    pedidoEntregadoDetalles.ShowDialog();
                }
                catch (Exception ex)
                {
                    MensajeError(ex.Message);
                }
            }
        }
    }
}
